api/src/iot/import_utils_apis.py

```diff
@@ -0,0 +1,484 @@
+from collections import Counter
+from typing import Generator, List
+
+import requests
+from django.conf import settings
+
+from iot import import_utils, models
+from iot.import_utils import LatLong, PersonData, SensorData
+
+API = 'https://maps.amsterdam.nl/open_geodata/geojson_lnglat.php?'
+API_MAPPER = {
+    'wifi_sensor_crowd_management': f'{API}KAARTLAAG=CROWDSENSOREN&THEMA=cmsa',
+    'camera_brug_en_sluisbediening': f'{API}KAARTLAAG=PRIVACY_BRUGSLUIS&THEMA=privacy',
+    # 'cctv_camera_verkeersmanagement': f'{API}KAARTLAAG=VIS&THEMA=vis',
+    # 'kentekencamera_reistijd': f'{API}KAARTLAAG=VIS&THEMA=vis',
+    # 'kentekencamera_milieuzone': f'{API}KAARTLAAG=VIS&THEMA=vis',
+    # 'ais_masten': f'{API}KAARTLAAG=PRIVACY_AISMASTEN&THEMA=privacy',
+    # 'verkeersonderzoek_met_cameras': f'{API}KAARTLAAG=PRIVACY_OVERIG&THEMA=privacy',
+    # 'beweegbare_fysieke_afsluiting': f'{API}KAARTLAAG=VIS_BFA&THEMA=vis',
+}
+
+
+def import_api_data(api_name: str) -> tuple:
+    """
+    takes the name of the api as a param, calls the url that belongs to it in the
+    API_MAPPER. when the data is fetched from the api, it will convert it to a dict
+    and pass it to the migrate_api_data together with the api_name.
+    A tuple will be returned that will contain the results.
+    """
+    try:
+        api_url = API_MAPPER.get(api_name, None)  # get the url that belongs to the api or None.
+        if not api_url:
+            err = f"unknown api_name {api_name}"
+            return (err,)
+
+        response = requests.get(url=api_url)
+
+        # check if response is 200 and content type is json, if not, return
+        if response.status_code != 200 or \
+                'application/json' not in response.headers["Content-Type"]:
+            err = f"{response.status_code} - {response.content}"
+            return (err,)
+
+        data = response.json()  # get the content of the response as dict
+        return migrate_api_data(api_name=api_name, api_data=data)
+
+    except Exception as e:
+        err = f"{e}"
+        return (err,)
+
+
+def migrate_api_data(api_name: str, api_data: dict) -> tuple:
```

Comment:
Ik denk dat `convert_api_data` of iets dergelijks meer toepasselijk is. Migrate heeft normaal gesproken een andere betekenis binnen een django app

Reply:
ik heb het gewijzigd naar convert, maar ik denk dat convert is ook een misleading naam, want we zijn data aan het opslaan in de backend vanuit de api. ik dacht misschien aan populate_api_data.